### DISCUSSION TITLE
Try the new Focus View Preview!

### DISCUSSION THREAD

**Discussion by d13** [gitkraken/vscode-gitlens#2535] [Labs]
> 🚨 Heads-up!
> This feature has been moved to the [Launchpad](https://github.com/gitkraken/vscode-gitlens/discussions/3286)

We're excited to invite you try the preview of the all-new Focus View,  a ✨GitLens+ feature that landed in our 13.3.0 release.

![Fpq0hF4aUAARRCq](https://user-images.githubusercontent.com/293282/222561589-4a61ae17-9889-441b-9dbd-b2f9e4286afa.jpg)

The idea is to provide you with a comprehensive list of all your most important work across your connected GitHub repos.

This is still very early and we are currently still playing with additional features/discovery, more providers, colors, etc. We really want to hear your feedback -- what do you think, is this useful, what does it have that you like, what is it missing, etc.

To try this out, use `GitLens+: Show Focus View` from the command palette.

![Screen Shot 2023-03-02 at 4 41 48 PM](https://user-images.githubusercontent.com/293282/222562641-675066eb-2fed-4927-afd2-fc634a41fdc8.png)

Stay up to date with the latest changes to the Focus View by switching to the pre-release version of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![199397587-6948b1bc-3019-4d85-a6d5-332fea5d8f78](https://user-images.githubusercontent.com/293282/222563747-b38113c1-e33b-48b9-9036-538043b813a5.png)

**Comment by Zethtren** [comment]
It would be nice to add the arrows from the Issues and Pull Requests extension so I could click the issue and have it create & switch the active branch. Otherwise it is great! And this is, at least, easier than clicking through each repo and expanding out the issues to find what I need to work on!

**Comment by oguimbal** [comment]
My focus view is empty (private github repo)... even if there are supposed to be dozens of PRs here

**Reply by zalewskigrzegorz** [reply]
same
![image](https://github.com/gitkraken/vscode-gitlens/assets/12560152/2787028a-6708-4535-a8b3-15260dc9446c)

**Reply by jkelroy** [reply]
@oguimbal @zalewskigrzegorz the focus view will only show pull requests (or issues) opened by you, assigned to you, needing your review, or that mentions you. It will not show every pull request in the repository - `All` shows all pull requests from the four mentioned cases. To confirm, are pull requests that meet one of those cases not showing for you?

**Reply by zalewskigrzegorz** [reply]
@jkelroy I got the Idea in Kraken, I saw my PR. But not in VS
![image](https://github.com/gitkraken/vscode-gitlens/assets/12560152/62deca8f-d5ed-472f-ba09-6102ce718abd)

**Reply by jkelroy** [reply]
Do you have more than one remote added? If so, navigate to the [Remotes view](https://help.gitkraken.com/gitlens/side-bar/#remotes-view), right-click the remote where the issues/pull requests are located, and set it to the default. Do issues appear in the focus view after?
<img width="411" alt="Screenshot 2023-06-15 at 9 23 33 AM" src="https://github.com/gitkraken/vscode-gitlens/assets/27739784/fe4be4c4-51df-483b-a22a-8038b439744f">

**Reply by eamodio** [reply]
@zalewskigrzegorz are you using a private instance of GitHub too?

**Reply by zalewskigrzegorz** [reply]
Yeah, this is an internal/private project

**Reply by zalewskigrzegorz** [reply]
@jkelroy i had only one remote
![image](https://github.com/gitkraken/vscode-gitlens/assets/12560152/14476fbe-38bc-4d44-9775-8838a31687e0)

**Reply by jkelroy** [reply]
We have heard of other utilizing GitHub Enterprise having issues working with the focus view so I started an issue [here](https://github.com/gitkraken/vscode-gitlens/issues/2744). Can you please include your logs by running from _GitLens: Enable Debug Logging_ from the Command Palette (<kbd>F1</kbd> or <kbd>ctrl/cmd</kbd>+<kbd>shift</kbd>+<kbd>p</kbd>). This will enable logging to the _GitLens_ and _GitLens (Git)_ channel in the _Output_ pane.

Once enabled, please reproduce the issue, and attach the log from the both _GitLens_ and _GitLens (Git)_ channels on this issue.

Thank you!

**Comment by ifireball** [comment]
Would be nice to be able to hide the issues since I don't use them.

**Reply by jkelroy** [reply]
@ifireball I created a feature request to track this here #2820.

**Comment by rileychh** [comment]
When used in Traditional Chinese, the date column wraps to multiple lines, which doesn't look good.

![Screenshot](https://github.com/gitkraken/vscode-gitlens/assets/53939195/46186e32-ec4e-47fe-a159-d669509da7c3)

**Comment by NicolasRouquette** [comment]
When will the VSCode GitLens client have feature parity w.r.t. the GitKraken client w.r.t. integration support, particularly GitLab self hosted?

https://help.gitkraken.com/gitkraken-client/gitkraken-focus-view/#connect-integrations

**Comment by timothyfroehlich** [comment]
It looks like it will only show PRs that are "yours" in some way? There's a PR that I'm collaborating on that doesn't show up in any of the views because the author hasn't clicked the button to add me to the review yet. Can we just have an "all" filter?

<img width="120" alt="image" src="https://github.com/gitkraken/vscode-gitlens/assets/5819722/b3038c00-2837-4ac9-866a-54b868ff927f">

**Reply by FTOSevenK** [reply]
> It looks like it will only show PRs that are "yours" in some way? There's a PR that I'm collaborating on that doesn't show up in any of the views because the author hasn't clicked the button to add me to the review yet. Can we just have an "all" filter?
> 
> <img width="120" alt="image" src="https://github.com/gitkraken/vscode-gitlens/assets/5819722/b3038c00-2837-4ac9-866a-54b868ff927f">
>

**Comment by timothyfroehlich** [comment]
I'd opened it again because I knew I needed to do some reviews and I was looking at it and realizing that I don't know what I should do with any of them. The one that I need to review has no marking that shows that. So... it's kinda useless to me right now.
